include/glow/Backends/TraceEvents.h

```diff
@@ -161,6 +166,19 @@ class TraceContext {
   void
   logTraceEvent(llvm::StringRef name, llvm::StringRef type, uint64_t timestamp,
                 std::map<std::string, std::string> additionalAttributes = {});
+
+  /// Sets the human readable \p name for thread \tid.
+  void setThreadName(int tid, llvm::StringRef name);
+  /// \returns the list of human readable thread names.
+  std::map<int, std::string> &getThreadNames() { return threadNames_; }
+
+  /// Dumps all TraceEvents in json format to the given \filename, optionally
+  /// with a provided \p processName.
+  void dump(llvm::StringRef filename, std::string processName = "");
+
+  /// Moves all TraceEvents and thread names in \p other into this context.
```

Comment:
What is the use-case for the raw ptr version of merge? Is this to allow a TraceContext to be essentially emptied and then reused without needing to reallocate?
I'd extend the comment a bit to call out that it is destructive on `other`.

Reply:
Yes pretty much, allows reuse of TraceContext. Will add the comment.